third_party/intel/backend/driver.py

```diff
@@ -10,17 +12,61 @@
 ze_root = os.getenv("ZE_PATH", default="/usr/local")
 include_dir = [os.path.join(ze_root, "include")]
 
-oneapi_root = os.getenv("ONEAPI_ROOT")
-if oneapi_root:
-    include_dir += [
-        os.path.join(oneapi_root, "compiler/latest/include"),
-        os.path.join(oneapi_root, "compiler/latest/include/sycl")
-    ]
+
+def find_sycl(include_dir: list[str], library_dir: list[str]) -> tuple[list[str], list[str]]:
+    """
+    Looks for the sycl library in known places.
+
+    Arguments:
+      include_dir: list of include directories to pass to compiler.
+      library_dir: list of library directories to pass to compiler.
+
+    Returns:
+      enriched include_dir and library_dir.
+
+    Raises:
+      AssertionError: if library was not found.
+    """
+    assertion_error = lambda: AssertionError(
+        "sycl headers not found, please install icpx compiler, provide ONEAPI_ROOT environment "
+        "or install intel-sycl-rt>=2025.0.0 wheel")
+
+    if shutil.which("icpx"):
+        return include_dir, library_dir
+
+    oneapi_root = os.getenv("ONEAPI_ROOT")
+    if oneapi_root:
+        include_dir += [
+            os.path.join(oneapi_root, "compiler/latest/include"),
+            os.path.join(oneapi_root, "compiler/latest/include/sycl")
+        ]
+        return include_dir, library_dir
+
+    try:
+        sycl_rt = importlib.metadata.metadata('intel-sycl-rt')
+    except importlib.metadata.PackageNotFoundError:
+        raise assertion_error()
+
+    if tuple(map(eval, sycl_rt.get('version').split('.'))) < (2025, 0, 0):
+        raise assertion_error()
+
+    for f in importlib.metadata.files('intel-sycl-rt'):
+        # sycl/sycl.hpp and sycl/CL/sycl.hpp results in both folders
+        # being add: include and include/sycl.
+        if f.name == 'sycl.hpp':
+            include_dir += [os.path.abspath(f.locate().parent.parent)]
```

Comment:
Method `locate()` returns the absolute path, no need to call `abspath()`
```suggestion
            include_dir += [f.locate().parent.parent.as_posix()]
```

Reply:
abspath was used to get rid of relative path navigation like this:

```
In [12]: f.locate().parent.parent.as_posix()
Out[12]: '/home/jovyan/Projects/poc-wheel/.venv/lib/python3.9/site-packages/../../../include/sycl'

In [13]: os.path.abspath(f.locate().parent.parent)
Out[13]: '/home/jovyan/Projects/poc-wheel/.venv/include/sycl'
```

Reply:
What about
```
f.locate().parent.parent.absolute().as_posix()
```

IMHO it's better than mixing `os.path` and `pathlib` in the same expression

Reply:
```
f.locate().parent.parent.absolute().as_posix()
Out[15]: '/home/jovyan/Projects/poc-wheel/.venv/lib/python3.9/site-packages/../../../include/sycl'
```

Reply:
I found. There is a `resolve()` method